lib/cublas/linalg.jl

```diff
@@ -37,6 +37,18 @@ end
 LinearAlgebra.norm(x::DenseCuArray{<:CublasFloat}) = nrm2(x)
 LinearAlgebra.BLAS.asum(x::StridedCuArray{<:CublasFloat}) = asum(length(x), x)
 
+function LinearAlgebra.norm(x::DenseCuArray{<:CublasFloat}, p::Integer)
+    if p==1
+        return CUBLAS.asum(length(x),x)
+    end
+    if p==2
+        return LinearAlgebra.norm(x)
+    end
+    if p>2
+        return LinearAlgebra.tr(LinearAlgebra.Diagonal(abs.(x))^p)^(1/p)
```

Comment:
The problem is that these operations are implemented using CUBLAS, so are themselves restricted to the types CUBLAS supports (plain, basic C types). Ideally we'd have something more generic. What about https://github.com/JuliaGPU/CUDA.jl/issues/84#issuecomment-626571308, is that not valid or slower?

Reply:
Such a generic versions could also go into GPUArrays.jl, while a specialized version for CUBLAS-supported types could live in CUDA.jl.

Reply:
@maleadt When we use the method you mentioned in [#84](https://github.com/JuliaGPU/CUDA.jl/issues/84#issuecomment-626571308), it throws out a scalar indexing warning, which wouldn't work when `CUDA.allowscalar(false)` is asserted. That is why I have implemented it using CUBLAS based operations.

Reply:
@maleadt Could you elaborate on the types CUBLAS does not support?

Reply:
`mapreduce` is supported by CUDA.jl, I don't understand how that would trigger scalar indexing?

If you look at the CUBLAS docs, you'll see it's a C library that only supports a limited set of element types. That's why we have type unions like CUBLASFloat in CUDA.jl